packages/airnode-node/src/evm/fulfillments/api-calls.test.ts

```diff
@@ -526,8 +260,1048 @@ describe('submitApiCall', () => {
+        failMock.mockRejectedValueOnce(failTxError);
+        const apiCall = fixtures.requests.buildSuccessfulApiCall({
+          id: '0xb56b66dc089eab3dc98672ea5e852488730a8f76621fd9ea719504ea205980f8',
+          data: {
+            encodedValue: '0x448b8ad3a330cf8f269f487881b59efff721b3dfa8e61f7c8fd2480389459ed3',
+            signature:
+              '0xda6d5aa27f48aa951ba401c8a779645f7d1fa4a46a5e99eb7da04b4e059449a834ca1058c85dfe8117305265228f8cf7ae64c3ef3c4d1cc191f77807227dac461b',
+          },
+          nonce: 5,
+        });
+        const [logs, err, data] = await apiCalls.testAndSubmitFulfill(createAirnodeRrpFake(), apiCall, {
+          gasTarget,
+          contracts,
+          masterHDNode,
+          provider,
+        });
+        expect(logs).toEqual([
+          { level: 'DEBUG', message: `Attempting to fulfill API call for Request:${apiCall.id}...` },
+          {
+            error: staticCallError,
+            level: 'ERROR',
+            message: `Static call fulfillment failed for Request:${apiCall.id} with ${staticCallError}`,
+          },
+          { level: 'INFO', message: `Submitting API call fail for Request:${apiCall.id}...` },
+          {
+            error: failTxError,
+            level: 'ERROR',
+            message: `Error submitting API call fail transaction for Request:${apiCall.id}`,
+          },
+        ]);
+        expect(err).toEqual(failTxError);
+        expect(data).toEqual(null);
+        expect(staticFulfillMock).toHaveBeenCalledTimes(2);
+        expect(staticFulfillMock).toHaveBeenNthCalledWith(
+          2,
+          apiCall.id,
+          apiCall.airnodeAddress,
+          apiCall.fulfillAddress,
+          apiCall.fulfillFunctionId,
+          '0x448b8ad3a330cf8f269f487881b59efff721b3dfa8e61f7c8fd2480389459ed3',
+          '0xda6d5aa27f48aa951ba401c8a779645f7d1fa4a46a5e99eb7da04b4e059449a834ca1058c85dfe8117305265228f8cf7ae64c3ef3c4d1cc191f77807227dac461b',
+          txOpts
+        );
+        expect(fulfillMock).not.toHaveBeenCalled();
+        expect(failMock).toHaveBeenCalledTimes(2);
+        expect(failMock).toHaveBeenNthCalledWith(
+          2,
+          apiCall.id,
+          apiCall.airnodeAddress,
+          apiCall.fulfillAddress,
+          apiCall.fulfillFunctionId,
+          'Static call error',
+          txOpts
+        );
+      });
+    });
+  });
+
+  describe('estimateGasAndSubmitFulfill', () => {
+    describe('Pending API calls', () => {
+      test.each([gasTargetWithoutGasLimit, gasTargetFallbackWithoutGasLimit])(
+        `successfully estimates gas and submits a fulfill transaction for pending requests - %#`,
+        async (gasTarget) => {
+          const txOpts = { ...gasTarget, nonce: 5 };
+          const provider = new ethers.providers.JsonRpcProvider();
+          estimateAirnodeRrpOverheadMock.mockResolvedValueOnce(ethers.BigNumber.from(73804));
+          estimateFulfillmentCallOverheadMock.mockResolvedValueOnce(ethers.BigNumber.from(290001));
```

Comment:
Maybe variables for `73804` and `290001`?

Reply:
Because they are originally ethers.BigNumber, I'm summing them using `.add` method as in [here](https://github.com/api3dao/airnode/blob/64afdf08addbb32b12ed9e5b94047c301a09cb0a/packages/airnode-node/src/evm/fulfillments/api-calls.ts#L319). So using primitive numbers for variables causes error unless we mock them.